includes/class-modal-checkout.php

```diff
@@ -489,6 +489,12 @@ public static function wc_get_template( $located, $template_name ) {
 			}
 		}
 
+		// This is for the initial display – the markup will be refetched on cart updates (e.g. applying a coupon).
+		// Then it'd be handled by the `woocommerce_update_order_review_fragments` filter.
+		if ( 'checkout/review-order.php' === $template_name && ! self::should_show_order_details() ) {
+			$located = NEWSPACK_BLOCKS__PLUGIN_DIR . 'src/modal-checkout/templates/empty-order-details.php';
```

Comment:
```suggestion
			ob_start();
			echo '<table class="shop_table woocommerce-checkout-review-order-table empty"></table>';
			return ob_end_flush();
```

Non-blocking: we can remove the need for a separate file for the empty table template by using an output buffer.